lib/src/components/NxDropdown/stateful/__tests__/NxStatefulDropdown.test.tsx

```diff
@@ -5,192 +5,311 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { act } from 'react-dom/test-utils';
-import { mount, ReactWrapper } from 'enzyme';
-import 'jest-enzyme';
-import * as enzymeUtils from '../../../../__testutils__/enzymeUtils';
 
-import NxButton from '../../../NxButton/NxButton';
-import NxDropdown from '../../NxDropdown';
+import { render, screen, fireEvent, within, createEvent } from '@testing-library/react';
+import { rtlRenderElement, rtlRender, userEvent } from '../../../../__testutils__/rtlUtils';
+
 import NxStatefulDropdown, { Props } from '../NxStatefulDropdown';
 
 describe('NxStatefulDropdown', () => {
-  let container: HTMLDivElement | null;
-
   const minimalProps = {
-        label: 'dropdown',
-        children: <a>Hello</a>
-      },
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxStatefulDropdown, minimalProps),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxStatefulDropdown, minimalProps);
-
-  beforeEach(function() {
-    // Avoid rendering directly on the body.
-    container = document.createElement('div');
-    document.body.appendChild(container);
-  });
-
-  afterEach(function() {
-    if (container) {
-      document.body.removeChild(container);
-      container = null;
-    }
-  });
+    label: 'dropdown-toggle',
+    children: <a>Hello</a>
+  };
 
-  it('renders an NxDropdown component', function() {
-    const component = getShallowComponent();
-    expect(component).toMatchSelector(NxDropdown);
-    expect(component).toHaveProp('label', 'dropdown');
-  });
+  const renderEl = rtlRenderElement<Props>(NxStatefulDropdown, minimalProps);
+  const quickRender = rtlRender<Props>(NxStatefulDropdown, minimalProps);
 
-  it('renders an NxDropdown component passing on the given props', function() {
-    const component = getShallowComponent({
+  it('renders an NxStatefulDropdown toggle element correctly with the given props', function() {
```

Comment:
This seems like it's squashing together things that unrelated and tested separately on `NxDropdown` itself.  I'd want to see the tests here follow a similar pattern.  Don't follow the patterns present in the old `NxStatefulDropdown` tests, as those were designed around ensuring the that proper props are passed on to the child react component which is very different from what we're trying to test now.

Reply:
> This seems like it's squashing together things that are unrelated and tested separately on `NxDropdown` itself. I'd want to see the tests here follow a similar pattern. Don't follow the patterns present in the old `NxStatefulDropdown` tests, as those were designed around ensuring the that proper props are passed on to the child react component which is very different from what we're trying to test now.

I cleaned that up here: 041a235e97a4bd79499e548d3229ff2fbfaf70db

but realized there was already a disabled test below so I removed it and cleaned it up here:
5a150c1b2afd48ee1e108923be99f720bfef7a24